### PR TITLE
 Introducing searchlight_1 Branch

### PR DESCRIPTION
**Pull request by saramazz**
Here's a summary of the changes in searchlight_1:

1) Updates in BayesianAnalysis.py:
   - Renamed functions and split into two separate files: one for run_bayesian_analysis_router and one for its version applied to coordinates and areas.
   - Removed plot_distribution_statistics() and standardized result saving (it occours onlt in run_bayesian_analysis_router.

2) Changes in MainScript.py:
   - Updated imports of the functions from other files.
   - Results are now saved only within the run_bayesian_analysis_* function.

These changes aim to improve code organization and functionality within the branch

### CODE REVIEW

File: neuroinfer/code/MainScript.py [hunk 1]
```diff
@@ -1,39 +1,102 @@
 # MainScript.py
 import argparse
-from neuroinfer.code.BayesianAnalysis import run_bayesian_analysis
+from neuroinfer.code.BayesianAnalysis import *
```
Comment (mlionello) [inline neuroinfer/code/MainScript.py]: better to specify individually the functions imported

File: neuroinfer/code/run_bayesian.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
+
```
Comment (mlionello) [inline neuroinfer/code/run_bayesian.py:16]: import missing?

File: neuroinfer/code/dimension_json.py
Comment (mlionello) [inline neuroinfer/code/dimension_json.py:1]: into script subfolder

File: neuroinfer/code/results.py
Comment (mlionello) [inline neuroinfer/code/results.py:1]: into script subfolder

**Review by mlionello — APPROVED** [review]

**Review by davidecrc — APPROVED** [review]

**Pull request merged**